lambda/matching.ml

```diff
@@ -1466,13 +1576,14 @@ let get_mod_field modname field =
   lazy (
     let mod_ident = Ident.create_persistent modname in
     let env = Env.add_persistent_structure mod_ident Env.initial_safe_string in
+    let loc = Location.in_file !Location.input_name in
```

Comment:
I've actually left a note about that function in my initial round of review, unfortunately as the code disappeared github gives me a link that 404s, so let me reproduce my comment:

> This function looks a lot like Lambda.transl_prim, I actually think that it existence is a mistake at this point.
> 
> However, it's interesting to note that while you update this function, you didn't update transl_prim. Why is that?
> Also, I'm not sure if this change is actually valuable, or even correct.

As a result, it looks like @mshinwell instead of removing this function and using `Lambda.transl_prim` just updated `transl_prim` in the same way.
I still think they should be deduplicated, the difference between the two is in which environment they use (`initial_safe_string` or `empty`) and I believe it is meaningless in that context.

As for whether the location is correct, I don't think so. Is it any better than `Location.none`? I doubt it.
As for whether we can do better, without too much pain, I'll let you guys decide this.

Reply:
@trefis and I look at this and concluded that it isn't clear the `lazy` is valuable here in any case, so it's been removed.  The locations have been fixed not to use the reference here.